actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -53,6 +59,8 @@ class ActionCable.Connection
     @isState("open", "connecting")
 
   # Private
+  isProtocolSupported: ->
+    @getProtocol() isnt unsupportedProtocol and @getProtocol() in supportedProtocols
```

Comment:
Isn't checking `@getProtocol() in supportedProtocols` enough? If the protocol is in that array then it's not `unsupportedProtocol`.